sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/FetchSubscriberDbConnector.groovy

```diff
@@ -0,0 +1,103 @@
+
+    FetchSubscriberDbConnector(ConnectionProvider connectionProvider){
+        this.connectionProvider = connectionProvider
+    }
+
+    @Override
+    Map<String, Status> getNotificationsForDay(LocalDate day){
+        Instant startOfDay = day.atStartOfDay().atZone(ZoneId.systemDefault()).toInstant()
+        Instant endOfDay = day.plusDays(1).atStartOfDay().minusNanos(1).atZone(ZoneId.systemDefault()).toInstant()
+        String sqlQuery = SELECT_NOTIFICATIONS + " WHERE arrival_time BETWEEN ? AND ?"
+        Map<String, Status> foundNotifications = new HashMap<>()
+
+        try{
+            Connection connection = connectionProvider.connect()
+
+            connection.withCloseable {
+                PreparedStatement preparedStatement = it.prepareStatement(sqlQuery)
+                preparedStatement.setTimestamp(1, Timestamp.from(startOfDay))
+                preparedStatement.setTimestamp(2, Timestamp.from(endOfDay))
+                preparedStatement.execute()
+
+                def resultSet = preparedStatement.getResultSet()
+                while (resultSet.next()) {
+                    String sampleCode = resultSet.getString("sample_code")
+                    Status status = Status.valueOf(resultSet.getString("sample_status"))
+                    foundNotifications.put(sampleCode,status)
+                }
+            }
+        }catch(Exception exception){
+            throw new DatabaseQueryException(exception.message)
+        }
+        return foundNotifications
+    }
+
+    @Override
+    List<Subscriber> getSubscriberIdForSamples(Map<String,Status> sampleToStatus){
+
+        List<Subscriber> subscribers = []
+        try{
+            Connection connection = connectionProvider.connect()
+
+            connection.withCloseable { Connection con ->
+                sampleToStatus.each {sampleEntry ->
+                    //get all subscriptions for a person, id to list of project codes
+                    String sqlQuery = JOIN_SUBSCRIBERS_SUBSCRIPTIONS + " WHERE ? LIKE CONCAT(project_code ,'%') "
+
+                    PreparedStatement preparedStatement = con.prepareStatement(sqlQuery)
+                    preparedStatement.setString(1,sampleEntry.key)
+                    preparedStatement.execute()
+                    def resultSet = preparedStatement.getResultSet()
+
+                    while(resultSet.next()){
+                        String firstName = resultSet.getString("first_name")
+                        String lastName = resultSet.getString("last_name")
+                        String email = resultSet.getString("email")
+
+                        if(subscribers.find {it.email == email}){
+                            subscribers.find{it.email == email}.subscriptions.put(sampleEntry.key,sampleEntry.value)
+                        }else{
+                            subscribers << new Subscriber(firstName,lastName,email,sampleToStatus)
+                        }
```

Comment:
I'm not sure if the else case should be implemented as such. Should we create new subscribers and if so should this be done in this method or should this be done seperately.

Reply:
Ok, so I talked to @KochTobi and we found indeed, that this should not be part of the DB. The transformation from samples to subscriber is rather a domain thing. I will update that :)